addons/binding/org.openhab.binding.nuki/ESH-INF/thing/thing-types.xml

```diff
@@ -0,0 +1,60 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<thing:thing-descriptions bindingId="nuki"
+        xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xmlns:thing="http://eclipse.org/smarthome/schemas/thing-description/v1.0.0"
+        xsi:schemaLocation="http://eclipse.org/smarthome/schemas/thing-description/v1.0.0 http://eclipse.org/smarthome/schemas/thing-description-1.0.0.xsd">
+
+    <!-- Nuki Bridge (Bridge Type) -->
+    <bridge-type id="bridge">
+        <label>Nuki Bridge</label>
+        <description>This bridge represents a Nuki Bridge on your local network. Nuki Smart Locks have to be paired via Bluetooth with it.</description>
+        <config-description>
+            <parameter name="ip" type="text" required="true">
+                <label>IP Address</label>
+                <description>The IP Address of the Nuki Bridge.</description>
+            </parameter>
+            <parameter name="port" type="integer" required="true">
+                <label>Port</label>
+                <description>The Port on which the Nuki Bridge REST API runs.</description>
+                <default>8080</default>
+            </parameter>
+            <parameter name="apiToken" type="text" required="true">
+                <label>API Token</label>
+                <description>The API Token configured via the Nuki App when enabling the API/IFTTT.</description>
+            </parameter>
+        </config-description>
+    </bridge-type>
+
+    <!-- Nuki Smart Lock (Thing Type) -->
+    <thing-type id="smartLock">
+        <supported-bridge-type-refs>
+            <bridge-type-ref id="bridge"/>
+        </supported-bridge-type-refs>
+        <label>Nuki Smart Lock</label>
+        <description>Nuki Smart Lock which is paired via Bluetooth to a Nuki Bridge.</description>
+        <channels>
+            <channel id="smartLockOpenClose" typeId="channelTypeSmartLockOpenClose"/>
+            <channel  id="smartLockUnlatchClose" typeId="channelTypeSmartLockUnlatchClose"/>
+        </channels>
+        <config-description>
+            <parameter name="nukiId" type="text" required="true">
+                <label>Nuki ID</label>
+                <description>The ID of the Nuki Smart Lock.</description>
+            </parameter>
+        </config-description>
+    </thing-type>
+
+    <!-- Channel Type -->
+    <channel-type id="channelTypeSmartLockOpenClose">
+        <item-type>Switch</item-type>
+        <label>Nuki Smart Lock Open/Close</label>
+        <description>Nuki Smart Lock channel to open or close the door.</description>
+    </channel-type>
+
+    <channel-type id="channelTypeSmartLockUnlatchClose">
+        <item-type>Switch</item-type>
+        <label>Nuki Smart Lock Unlatch/Close</label>
+        <description>Nuki Smart Lock channel to unlatch or close the door.</description>
```

Comment:
> Well, you would like to dig deeper, ok :)

Oh hell, this is again a good example, why there is no way to come up with some "standard" :-) In this context, I [just read this](http://stadt-bremerhaven.de/warum-nuki-fuer-alexa-keine-smart-home-komponente-ist/):
> Das Smart Home Produkt von Amazon ist leider für europäische Türschlösser nicht geeignet, da es ausschließlich für US-Türschlösser konzipiert wurde (bei Apple Homekit droht dasselbe Problem). Dies hat zur Folge, dass die Funktion „Tür öffnen“, welche auch die Falle des Türschlosses ziehen kann, im Smart Home Skill nicht vorhanden ist (weil US-Schlösser das in dieser Form nicht haben).

> I am planning to implement: ...

This feels like far too many channels to me. Let me try to give my rule of thumb for device modelling:
- Channels should be designed around possible features/characteristics/capabilities of the device. If the state of two such features cannot change independently from each other, they should be modelled as a single channel. It is ok to have more commands understood by such a channel then it has possible states - so you can at any time easily add a command.
- A Thing should only have channels that are indeed available for the user's setup. So it it better to define different potential channel types and then either dynamically add the appropriate channel or to leave it to the user to add it. It is also possible to introduce a configuration value for the Thing on which it is then decided, which channels it has.
- Channel modelling should not be made with a mapping to UI widgets in mind; Sitemaps give you the chance to define any kind of command for a button and if we need more, that is rather something that we should make possible for all bindings within the framework.

Following these rules, I would actually think that a single channel (plus 5+6) would be enough for the lock - because there is the single "lock state". But I see your point that with the current framework possibilities, your channel 3 should rather be "advanced" and you want to provide switch channels for a simpler usage and UI mapping.
So my suggestion would be to reduce 1+2 to just one channel and have some Thing configuration possibility, which would let the user define whether the lock should be unlocked or unlatched upon operation and whether it should use lock'n'go or not.
The users that want more than that should imho resort to the advanced channel 3.

Would that be a compromise? Note that I am just trying to avoid channel proliferation as much as possible...

P.S.: I see that we are lacking something like "commandOptions" besides "stateOptions" within the channel definitions. This is definitely something we should look at...